src/stream_base.h

```diff
@@ -214,6 +214,8 @@ class StreamResource {
   // Return 0 for success and a libuv error code for failures.
   virtual int DoTryWrite(uv_buf_t** bufs, size_t* count);
   // Perform a write of data, and either call req_wrap->Done() when finished
+  //      call when? can we call sync? "call Done() and return" suggests
+  //      first Done, then return, but probably not right
```

Comment:
We’re not strict about this – basically, your options are:

- return `0`. This means you can call `Done()` at any point, which is almost always asynchronous but might not need to be.
- return an error code.